twenty-first/src/shared_math/x_field_element.rs

```diff
@@ -131,7 +131,8 @@ impl XFieldElement {
 
     pub fn sample(digest: &Digest) -> Self {
         let elements = digest.values();
-        XFieldElement::new([elements[0], elements[1], elements[2]])
+        XFieldElement::new([elements[2], elements[3], elements[4]])
+        // XFieldElement::new([elements[0], elements[1], elements[2]])
```

Comment:
This change currently results in a clockcycle improvement of $5729 \rightarrow 5209$ for `sample_weights(130)`.

```rust
// Current: _ s4 s3 s2 s1 s0 addr i _w4 _w3 w2 w1 w0

// If we instead of picking sample := [w0, w1, w2], we pick sample := [w2, w3, w4]... we save 8 clockcycles per sample:

pop pop                        // _ s4 s3 s2 s1 s0 addr i w4 w3 w2

// This works with no prior change to `XFieldElement::sample()`:

// swap3 pop                   // _ s4 s3 s2 s1 s0 addr i _w4 w0 w2 w1
// swap3 pop                   // _ s4 s3 s2 s1 s0 addr i w1 w0 w2
// swap2                       // _ s4 s3 s2 s1 s0 addr i w2 w0 w1
// swap1                       // _ s4 s3 s2 s1 s0 addr i w2 w1 w0
```